contracts/GAUC.sol

```diff
@@ -0,0 +1,364 @@
+            msg.sender,
+            address(_token),
+            _amount,
+            _borrower,
+            _auctionEndTimestamp,
+            _maxFaceValue,
+            _minBidIncrement,
+            _maturity,
+            _price,
+            address(wallet)
+        );
+    }
+
+    function createERC721Auction(
+        IERC721 _token,
+        uint256[] calldata _ids,
+        address _borrower,
+        uint256 _auctionEndTimestamp,
+        uint256 _maxFaceValue,
+        uint256 _minBidIncrement,
+        uint256 _maturity,
+        uint256 _price
+    ) public override {
+        // TODO: deploy proxy
+        GSDIWallet wallet = new GSDIWallet();
+        wallet.initialize(address(gsdiNFT), address(this));
+
+        for (uint8 i = 1; i <= _ids.length; i++) {
+            _token.safeTransferFrom(msg.sender, address(wallet), _ids[i], "");
+        }
+
+        uint256 auctionId = _auctionIdTracker.current();
+
+        auctionInfo[auctionId] = AuctionInfo(
+            _auctionEndTimestamp,
+            _maxFaceValue,
+            _maturity,
+            _price,
+            _minBidIncrement,
+            address(wallet),
+            address(0),
+            _borrower,
+            AUCTION_STATUS.OPEN
+        );
+
+        _auctionIdTracker.increment();
+
+        emit ERC721AuctionCreate(
+            msg.sender,
+            address(_token),
+            _ids,
+            _borrower,
+            _auctionEndTimestamp,
+            _maxFaceValue,
+            _minBidIncrement,
+            _maturity,
+            _price,
+            address(wallet)
+        );
+    }
+
+    function cancel(uint256 _auctionId) public override {
+        AuctionInfo storage auction = updateAuctionStatus(_auctionId);
+
+        require(
+            auction.auctionStatus == AUCTION_STATUS.OPEN ||
+                auction.auctionStatus == AUCTION_STATUS.EXPIRED,
+            "GAUC: auction not open or expired"
+        );
+        require(auction.lowestBidder == address(0), "GAUC: bidder exists");
+
+        GSDIWallet(auction.IGSDIWallet).setExecutor(auction.borrower);
+        auction.auctionStatus = AUCTION_STATUS.CANCELED;
+
+        emit AuctionCancel(msg.sender, _auctionId);
+    }
+
+    function bid(uint256 _auctionId, uint256 _amount) public override {
+        AuctionInfo storage auction = updateAuctionStatus(_auctionId);
+        uint256 purchasePrice = getPurchasePrice(auction.price);
+
+        require(
+            auction.auctionStatus == AUCTION_STATUS.OPEN,
+            "GAUC: auction not open"
+        );
+        require(
+            auction.minBidIncrement <= _amount,
+            "GAUC: must be at least minBidIncrement"
+        );
+        require(
+            auction.lowestBid.mul(99) >= _amount.mul(100),
+            "GAUC: must be 1% lower than lowest bid"
+        );
+        require(
+            msg.sender == auction.lowestBidder ||
+                balanceAvailable(msg.sender) >= purchasePrice,
```

Comment:
Should just be `balanceAvailable(msg.sender) >= purchasePrice`, the additional feature for the lowest bidder to bid on the auction they are winning is not required.